### PR TITLE
chore(rslint_parser): support for multiple suites when bechmarking the parser

### PR DESCRIPTION
**Pull request by xunilrj**
Allow multiple suites when benchmarking the parser and the formatter.

```
> cargo bench_parser --help
cargo bench
Benchmark parser and formatter.
USAGE:
    cargo bench [option]
OPTIONS
    --save-baseline     Allows different runs to be compared.
    --feature           Possible values: parser, formatter
    --criterion=[true/false]  Run a series of statistical test to assess with the this run is faster or slower than previous runs.
    --suites=<IDS>      Runs the specified benchmarks. Use comma as separator.
                        Valid values are:
                            *: will run all benchmarks;
                            js: will benchmark all javascript libraries;
                            ts: will benchmark all typescript libraries;
                        Default is "*".
    --filter=<file>     Filters out tests that don't match the query.
    --help              Prints this help.
```

These are all the libraries:

```
Summary
-------
jquery.min.js, Total Time: 19.3173ms, tokenization: 3.3534ms, parsing: 7.4134ms, tree_sink: 8.5505ms
vue.global.prod.js, Total Time: 25.9847ms, tokenization: 3.6636ms, parsing: 7.8456ms, tree_sink: 14.4755ms
react.production.min.js, Total Time: 1.1472ms, tokenization: 157.3µs, parsing: 345.9µs, tree_sink: 644µs
react-dom.production.min.js, Total Time: 20.8793ms, tokenization: 3.0538ms, parsing: 6.8883ms, tree_sink: 10.9372ms
dojo.js, Total Time: 5.1406ms, tokenization: 993.1µs, parsing: 1.3101ms, tree_sink: 2.8374ms
d3.min.js, Total Time: 55.1201ms, tokenization: 7.4085ms, parsing: 18.4736ms, tree_sink: 29.238ms
pixi.min.js, Total Time: 68.2168ms, tokenization: 9.7529ms, parsing: 23.2392ms, tree_sink: 35.2247ms
three.min.js, Total Time: 75.6594ms, tokenization: 11.101ms, parsing: 23.6065ms, tree_sink: 40.9519ms
tex-chtml-full.js, Total Time: 153.9711ms, tokenization: 22.7566ms, parsing: 45.8523ms, tree_sink: 85.3622ms
ios.d.ts, Total Time: 146.782ms, tokenization: 28.2088ms, parsing: 38.3091ms, tree_sink: 80.2641ms
checker.ts, Total Time: 142.2932ms, tokenization: 27.4954ms, parsing: 32.3618ms, tree_sink: 82.436ms
parser.ts, Total Time: 4.8313ms, tokenization: 1.8577ms, parsing: 1.0932ms, tree_sink: 1.8804ms
router.ts, Total Time: 2.6621ms, tokenization: 520.9µs, parsing: 640.7µs, tree_sink: 1.5005ms
```

### CODE REVIEW

File: .github/workflows/bench_parser.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
         run: cargo build --release --locked -p xtask_bench
 
       - name: Run Bench on PR Branch
-        run: cargo bench_parser --save-baseline pr
+        run: cargo bench_parser --suites=* --save-baseline pr
```
Comment (MichaReiser) [inline .github/workflows/bench_parser.yml:53]: Can we default to run all suites if none is provided? That would improve the economics of the CLI a bit

File: xtask/bench/src/lib.rs [hunk 2]
```diff
@@ -71,9 +72,32 @@ fn err_to_string<E: std::fmt::Debug>(e: E) -> String {
...
+    let suites_to_run = suites_to_run.split(',');
+    for suite in suites_to_run {
+        match suite {
+            "*" => {
+                libs.extend(suites["js"].lines());
+                libs.extend(suites["ts"].lines());
+            }
+            "js" => libs.extend(suites["js"].lines()),
+            "ts" => libs.extend(suites["ts"].lines()),
+            _ => {}
```
Comment (MichaReiser) [inline xtask/bench/src/lib.rs:98]: We should emit at least a warning that this is an invalid suite name

File: xtask/bench/src/lib.rs [hunk 3]
```diff
@@ -71,9 +72,32 @@ fn err_to_string<E: std::fmt::Debug>(e: E) -> String {
     format!("{:?}", e)
 }
 
-pub fn run(filter: String, criterion: bool, baseline: Option<String>, feature: FeatureToBenchmark) {
+pub fn run(
+    filter: String,
+    criterion: bool,
+    baseline: Option<String>,
+    feature: FeatureToBenchmark,
+    suites_to_run: &str,
```
Comment (ematipico) [inline xtask/bench/src/lib.rs:80]: At this point it might make sense to have a struct here, the arguments are becoming a lot

**Review by ematipico — APPROVED** [review]

File: .github/workflows/bench_parser.yml (hunk 1 above)
Comment (xunilrj) [reply .github/workflows/bench_parser.yml:53]: This is actually already done. But I will change the github action here to make it more explicit.

Comment (xunilrj) [reply .github/workflows/bench_parser.yml]: Done.

File: xtask/bench/src/lib.rs (hunk 2 above)
Comment (xunilrj) [reply xtask/bench/src/lib.rs]: Done.

File: xtask/bench/src/lib.rs (hunk 3 above)
Comment (xunilrj) [reply xtask/bench/src/lib.rs]: Done.

**Pull request merged**